### PR TITLE
Bump dependency versions to avoid conflicts

### LINKED ISSUE #53761: ansible azure requirements are self-conflicting
##### SUMMARY
``packaging/requirements/requirements-azure.txt`` cannot even be installed alone without causing conflicts between its own list of packages.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
packaging

##### ANSIBLE VERSION
master, also appling to older versions and breaking the use of azure extra: `pip install ansible[azure]`

##### STEPS TO REPRODUCE

```bash
pip install packaging/requirements/requirements-azure.txt
```
##### EXPECTED RESULTS
``pip check`` should never report conflicts after installing the requriments, this check **must be part of the CI testing** to avoid ever publishing conflicting requrements.

##### ACTUAL RESULTS

```
pip check
azure-mgmt-web 0.32.0 has requirement msrestazure~=0.4.7, but you have msrestazure 0.5.0.
azure-mgmt-keyvault 0.40.0 has requirement msrestazure~=0.4.7, but you have msrestazure 0.5.0.
azure-mgmt-batch 4.1.0 has requirement msrestazure~=0.4.7, but you have msrestazure 0.5.0.
```

### PR DESCRIPTION
**Pull request by jborean93**
##### SUMMARY
The `msrestazure==0.5.0` fails to be met the pinned versions of `azure-mgmt-batch`, `azure-mgmt-keyvault` and `azure-mgmt-web`. This PR updates those dependencies so `pip install ansible[azure]` works again.

Fixes https://github.com/ansible/ansible/issues/53761, although we should still fix CI to fail if that ever breaks again.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
Azure

### CODE REVIEW

**Comment by Fred-sun** [comment]
@zikalino need you review

**Review by ssbarnea — APPROVED** [review]
OK for me, we should approve once another Azure user confirms this works.

Personally I would have preferred to see a migration towards using ranges for requirements, but that can be done in another PR.

**Review by zikalino — APPROVED** [review]

**Pull request merged**
